File: app/src/main/java/torille/fi/lurkforreddit/data/models/jsonResponses/RedditToken.java
package torille.fi.lurkforreddit.data.models.jsonResponses;

import android.support.annotation.Nullable;

import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

/**
 * Model containing Reddit API auth token
 */


@AutoValue
public abstract class RedditToken {
    @SerializedName("access_token")
    public abstract String access_token();

    @SerializedName("token_type")
    public abstract String token_type();

    @SerializedName("scope")
    public abstract String scope();

    @Nullable
    @SerializedName("refresh_token")
    public abstract String refresh_token();

    @SerializedName("expires_in")
    public abstract long expires_in();

    public static TypeAdapter<RedditToken> typeAdapter(Gson gson) {
        return new AutoValue_RedditToken.GsonTypeAdapter(gson);

    }
}

File: app/src/main/java/torille/fi/lurkforreddit/data/models/jsonResponses/RedditToken.kt
package torille.fi.lurkforreddit.data.models.jsonResponses

import com.google.gson.annotations.SerializedName

/**
 * Model containing Reddit API auth token
 */


data class RedditToken(
        @SerializedName("access_token")
        val accessToken: String,
        @SerializedName("token_type")
        val tokenType: String,
        @SerializedName("scope")
        val scope: String,
        @SerializedName("refresh_token")
        val refreshToken: String?,
        @SerializedName("expires_in")
        val expiresIn: Long
)
